sm6375.c

```diff
@@ -0,0 +1,243 @@
+// SPDX-License-Identifier: BSD-3-Clause
+/* Copyright (c) 2022, Linaro Limited */
+
+#include <sys/mman.h>
+#include <err.h>
+#include <fcntl.h>
+#include <stdio.h>
+#include <stdint.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+
+#include "debugcc.h"
+
+static struct debug_mux cpu_cc = {
+	.phys = 0xfaa0000,
+	.size = 0x1000,
+
+	.enable_reg = 0x18,
+	.enable_mask = BIT(0),
+
+	.mux_reg = 0x18,
+	.mux_mask = 0x7f << 4,
+	.mux_shift = 4,
+
+	.div_reg = 0x18,
+	.div_mask = 0xf << 11,
+	.div_shift = 11,
+	.div_val = 4, // 8 for perfcl_clk
+};
+
+static struct debug_mux disp_cc = {
+	.phys = 0x5f00000,
+	.size = 0x20000,
+
+	.enable_reg = 0x3004,
+	.enable_mask = BIT(0),
+
+	.mux_reg = 0x7000,
+	.mux_mask = 0xff,
+
+	.div_reg = 0x3000,
+	.div_mask = 0xf,
+	.div_val = 4,
+};
+
+static struct debug_mux gcc = {
+	.phys = 0x1400000,
+	.size = 0x1f0000,
+
+	.enable_reg = 0x30004,
+	.enable_mask = BIT(0),
+
+	.mux_reg = 0x62000,
+	.mux_mask = 0x3ff,
+
+	.div_reg = 0x30000,
+	.div_mask = 0xf,
+	.div_val = 1,
+
+	.xo_div4_reg = 0x28008,
+	.debug_ctl_reg = 0x62038,
+	.debug_status_reg = 0x6203c,
+};
+
+static struct debug_mux gpu_cc = {
+	.phys = 0x5990000,
+	.size = 0x9000,
+
+	.enable_reg = 0x1100,
+	.enable_mask = BIT(0),
+
+	.mux_reg = 0x1568,
+	.mux_mask = 0xff,
+
+	.div_reg = 0x10fc,
+	.div_mask = 0xf,
+	.div_val = 2,
+};
+
+static struct measure_clk sm6375_clocks[] = {
+	{ "l3_clk", &gcc, 0xbf, &cpu_cc, 0x41 },
+	{ "perfcl_clk", &gcc, 0xbf, &cpu_cc, 0x25 },
+	{ "pwrcl_clk", &gcc, 0xbf, &cpu_cc, 0x21 },
+	{ "cpu_cc_debug_mux", &gcc, 0xbf },
```

Comment:
I think we usually comment out submux entries. Do you have any particular use case for them?

Reply:
This comment applies to other platforms from this patchset.

Reply:
I'll fix that!